src/Codeception/Lib/Connector/Yii2.php

```diff
@@ -301,6 +301,7 @@ public function doRequest($request)
         $queryString = parse_url($uri, PHP_URL_QUERY);
         $_SERVER['REQUEST_URI'] = $queryString === null ? $pathString : $pathString . '?' . $queryString;
         $_SERVER['REQUEST_METHOD'] = strtoupper($request->getMethod());
+        $_SERVER['QUERY_STRING'] = $queryString;
```

Comment:
It seems weird to me to store a value of type `string|null` here.. Should we not skip this line if `!isset($queryString)`?

Reply:
Sounds plausible. I searched for any 'specifications' on this, but couldn't really find them. When running a normal PHP script via Apache, the QUERY_STRING is an empty string when not set. But `parse_url` does indeed return null. I suspect an isset or maybe casting the null to a string is better?

Reply:
Let me do a quick check on nginx, sec...

Reply:
So, PHP built-in server omits it, nginx puts an empty string.
I agree; cast it to string!